resolwe/storage/connectors/registry.py

```diff
@@ -0,0 +1,61 @@
+"""Registry class for connectors"""
+import inspect
+
+from django.conf import settings
+
+
+class StorageConnectors:
+    """A class holding all configured instances of storage connectors.
+    The connector classes are registered automatically by the hook
+    __init_subclass__ in the BaseStorageConnector class when connector classes
+    are imported.
+
+    This class is implemented as singleton: constructing multiple instances
+    of it will always return the same instance.
+    """
+
+    __instance = None  #  A single instance of this class
+
+    @classmethod
+    def __new__(cls, *args, **kwargs):
+        if StorageConnectors.__instance is None:
+            StorageConnectors.__instance = super().__new__(cls)
+            StorageConnectors.__instance.__setattr__("_connectors", dict())
+        return StorageConnectors.__instance
+
+    def _add_storage_connector_class(self, connector):
+        """Add storage connector class to the registry. When there are
+        configuration entries in the settings for this class then
+        corresponding instances are created and stored in the registry.
+
+        If connector in not Class then exception is raised.
+        """
+
+        def _get_class_name(klass):
+            return "{}.{}".format(klass.__module__, klass.__name__)
+
+        assert inspect.isclass(connector)
+        full_class_name = _get_class_name(connector)
+        self._add_connectors_from_settings(full_class_name, connector)
+
+    def _add_connectors_from_settings(self, full_class_name, klass):
+        """Iterate through connectors defined in the settings and create
+        instance and add it to the registry if a match is found."""
+        for connector_name, connector_settings in settings.STORAGE_CONNECTORS.items():
+            if connector_settings["connector"] == full_class_name:
```

Comment:
I agree, but how? I could produce a warning when a class in the configuration is never matched but is will only be visible in logs. Maybe a management command that would check that? But somebody would have to run it manually every time configuration is changed.

Reply:
Appconfig can do validation in the `ready` method. Actually it would be nice if you transform this into two-steps operation - registry only collects connectors and then appconfig parses the configuration, validates it and initializes connectors. What's your opinion about that?

Reply:
I think that is a good idea. How should I report to user about unused configuration directive? Maybe by logging error, warning?

Reply:
I think logging error should do it.

Reply:
Done.